proposals/0001-lhls.md

```diff
@@ -0,0 +1,313 @@
+
+## Method of Advertising LHLS Segments
+
+#### In-Playlist Signaling
+
+In-playlist signaling happens when prefetch segments are advertised within the same playlist as complete segments. Prefetch segments are included at the bottom and are differentiated via the `EXT-X-PREFETCH` tag. This is the scheme outlined in this proposal. 
+
+We chose this scheme because it is backwards compatible with clients which have not implemented this proposal. According to the HLS specification clients must ignore any tags not outlined within; since `EXT-X-PREFETCH` is not an official tag clients will be left with a standard HLS playlist. This allows a server to deliver a single manifest which plays in both LHLS and unsupported clients. This scheme also allows clients to disable or enable LHLS while the client is streaming. Developers may make this option available to the user so that they are able to choose an experience most beneficial to them.
+
+Another benefit of this scheme is that it more easily takes advantage of the HLS specification. In most cases an in-playlist prefetch segments can be treated like a complete segments, which promotes code reuse and eases implementation. As an example a prefetch segment's Media Sequence Number is calculated relative to the complete segments in the playlist; if prefetch segments were elsewhere a new solution would have to be devised.
+
+#### All-Playlist Signaling
+
+All-Playlist signaling is when every Media Playlist contains only prefetch segments. In this scheme the `EXT-X-PREFETCH` tag is not required; all segments are assumed to be prefetch. This solution has the benefit of requiring little or no modifications from the HLS specification (what changes, if any, are hard to determine without actually making such a proposal). But this solution is not backwards compatible with clients not supporting this proposal.
+
+#### Separate-Playlist Signaling
+
+Separate-Playlist signaling is when a Master Playlist advertises an entire rendition as prefetch. Like all-playlist signaling, this scheme benefits from requiring little or no changes to the HLS specification on Media Playlists; it also has the added benefit of building backwards compatibility (contingent on an implementation which causes non-participating clients to ignore the rendition). This scheme, however, is more complex to implement in Hls.js. Hls.js works within the context of a single playlist - meaning that it chooses segments and gathers information from only one rendition. Supporting a second concurrent Media Playlist would most likely require significant amounts of new code as well as code refactored. Both servers and clients would also need to build a mechanism to synchronize properties such as Media Sequence Number and Discontinuity Sequence Numbers across complete and prefetch renditions.
```

Comment:
```suggestion
Separate-Playlist signaling is a strategy in which a Master Playlist advertises an entire rendition as prefetch. Like all-playlist signaling, this scheme benefits from requiring little or no changes to the HLS specification of Media Playlists; it also has the added benefit of building backwards compatibility (contingent on an implementation which causes non-participating clients to ignore the rendition). This scheme, however, is more complex to implement in Hls.js. Hls.js works within the context of a single playlist - meaning that it chooses segments and gathers information from only one rendition. Supporting a second concurrent Media Playlist would most likely require significant amounts of new code as well as code refactored. Both servers and clients would also need to build a mechanism to synchronize properties such as Media Sequence Number and Discontinuity Sequence Numbers across complete and prefetch renditions.
```
I think the main downside of this approach is that it complicates managing renditions on both the client side and server side. For example, logic is required to handle the case where you only have a non-prefetch version of one of the renditions (Is it ignored?).

Reply:
That's a good question. The client would just drop to regular latency mode in that case, but I don't see why an encoder would choose not to encode all of them as LHLS.